client/js/src/client.ts

```diff
@@ -36,6 +36,16 @@ export class NodeBlob extends Blob {
 	}
 }
 
+if (typeof window === "undefined") {
+	import("eventsource")
+		.then((EventSourceModule) => {
+			global.EventSource = EventSourceModule.default as any;
+		})
+		.catch((error) =>
+			console.error("Failed to load EventSource module:", error)
+		);
+}
+
```

Comment:
Thinking about it, I don't think we want to do this. This is why we have the issues with "cannot set window.Websocket" previously. We should just do this in the `eventsource_factory` method and directly return the class.